app/components/Header.tsx

```diff
@@ -0,0 +1,99 @@
+"use client";
+import { Link as ChakraLink } from "@chakra-ui/next-js";
+import { Box, Grid, Stack, IconButton } from "@chakra-ui/react";
+import Image from "next/image";
+import { usePathname } from "next/navigation";
+import { useState } from "react";
+import styled from "@emotion/styled";
+
+export default function Header() {
+  const [hamburger, setHamburger] = useState(false);
+
+  const pathname = usePathname();
+
+  const handleHamburger = () => {
+    setHamburger(!hamburger);
```

Comment:
I believe it's best practice in React to use this syntax when you're updating state based on the previous state: `setHamburger((prevState) => !prevState);` i.e. using the functional update form to set state
State updates can be asynchronous, so `hamburger` might not always reflect the latest state. By passing a function, you make sure that you're working with the most recent state value, which is especially useful when state can change rapidly (e.g. a user continuously clicks and unclicks the hamburger icon)